OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs

```diff
@@ -78,7 +80,14 @@ public void StartGame(Arguments args)
 			var replay = args != null ? args.GetValue("Launch.Replay", null) : null;
 			if (!string.IsNullOrEmpty(replay))
 			{
-				Game.JoinReplay(replay);
+				var replayMeta = ReplayMetadata.Read(replay);
```

Comment:
Moving the `CheckReplayCompatibility` check immediately after this (and adding a null check inside it) should fix my crash and polish issue that @penev92 explained.